controllers/machine_controller.go

```diff
@@ -647,38 +647,70 @@ func (r *MachineReconciler) nodeToMachine(o client.Object) []reconcile.Request {
 	}
 
 	// Match by nodeName and status.nodeRef.name.
-	filters := []client.ListOption{
+	allFilters := []client.ListOption{
 		client.MatchingFields{clusterv1.MachineNodeNameIndex: node.Name},
 	}
+	annotationfilters := []client.ListOption{}
```

Comment:
These variables are a bit confusing while reading. If I understood correctly the code here, `allFilters` is `annotationFilters` + the `MachineNodeNameIndex`.

If that's the case, let's please rename the list of filters to `filters`, have the `MachineNodeNameIndex` in a lone variable called something like `nodeNameFilter` and append it only when necessary.